tests/application-gateway-tests/test/gateway/suite.go

```diff
@@ -0,0 +1,235 @@
+
+	proxyconfig2 "github.com/kyma-project/kyma/components/application-gateway/pkg/proxyconfig"
+
+	"github.com/kyma-project/kyma/tests/application-gateway-tests/test/gateway/testkit/proxyconfig"
+
+	"github.com/google/uuid"
+
+	serviceCatalogClient "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset/typed/servicecatalog/v1beta1"
+	"github.com/kubernetes-sigs/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kyma-project/kyma/tests/application-gateway-tests/test/gateway/mock"
+	"github.com/kyma-project/kyma/tests/application-gateway-tests/test/tools"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	restclient "k8s.io/client-go/rest"
+)
+
+const (
+	defaultCheckInterval         = 2 * time.Second
+	appGatewayHealthCheckTimeout = 45 * time.Second
+	gatewayConnectionTimeout     = 20 * time.Second
+	apiServerAccessTimeout       = 60 * time.Second
+)
+
+type TestSuite struct {
+	httpClient            *http.Client
+	k8sClient             *kubernetes.Clientset
+	namespaceClient       corev1.NamespaceInterface
+	secretClient          corev1.SecretInterface
+	secretCreator         *proxyconfig.SecretsCreator
+	serviceInstanceClient serviceCatalogClient.ServiceInstanceInterface
+	config                TestConfig
+	appMockServer         *mock.AppMockServer
+
+	serviceInstanceName string
+}
+
+func NewTestSuite(t *testing.T) *TestSuite {
+	config, err := ReadConfig()
+	require.NoError(t, err)
+
+	k8sConfig, err := restclient.InClusterConfig()
+	require.NoError(t, err)
+
+	coreClientset, err := kubernetes.NewForConfig(k8sConfig)
+	require.NoError(t, err)
+
+	appMockServer := mock.NewAppMockServer(config.MockServicePort)
+
+	secretClient := coreClientset.CoreV1().Secrets(config.GatewayNamespace)
+
+	secretsCreator := proxyconfig.NewSecretsCreator(secretClient)
+
+	svcCatalogueClientSet, err := serviceCatalogClient.NewForConfig(k8sConfig)
+	require.NoError(t, err)
+
+	serviceInstanceName := fmt.Sprintf("gateway-tests-instance-%s", uuid.New().String()[:4])
+
+	return &TestSuite{
+		httpClient:            &http.Client{},
+		k8sClient:             coreClientset,
+		namespaceClient:       coreClientset.CoreV1().Namespaces(),
+		secretClient:          secretClient,
+		secretCreator:         secretsCreator,
+		config:                config,
+		appMockServer:         appMockServer,
+		serviceInstanceClient: svcCatalogueClientSet.ServiceInstances(config.GatewayNamespace),
+		serviceInstanceName:   serviceInstanceName,
+	}
+}
+
+func (ts *TestSuite) Setup(t *testing.T) {
+	ts.WaitForAccessToAPIServer(t)
+
+	ts.appMockServer.Start()
+
+	err := ts.createNamespace()
+	require.NoError(t, err)
+
+	err = ts.createServiceInstance()
+	require.NoError(t, err)
```

Comment:
Maybe we can perform error checking inside those methods like you do in CheckApplicationGatewayHealth. I think it's more consistent.

Reply:
I kind of prefer not to pass `*testing.T` to functions that do not really need it, as it makes them easier to reuse in the future.